docs/index.md

```diff
@@ -1,3 +1,134 @@
+<!-- vim: set textwidth=110: -->
+
 # rocd
 
-Welcome to rocd documentation!
+We start from a small preview of a system as a whole to understand what is the role of `rocd` component. The
+purpose of the [Roc Toolkit](https://roc-streaming.org/toolkit/docs/about_project/overview.html) (further
+referred to as `roc`) is to implement real-time audio streaming over network.
+
+## Entities
+
+Here are the top-level components:
+
+**host**
+:   Computer where the `roc` instance is currently running on. Computer platform may be on of:
+
+    - PC (personal computer, usually a desktop device with monitor or a similar output device),
+    - Android,
+    - iOS (future plans),
+    - embedded (is actually the same as PC, but usually doesn't have a monitor).
+
+**rocd**
+:   Server component used for stream management and auto discovery over network. Contains a library and a
+    binary that wraps it in a daemon. A stand-alone library is needed for mobile versions with their specific
+    platform limitations (programming language, permission models, ...).
+
+**roc-cast**
+:   Component used for two primary purposes:
+
+    - Provide UI for user interaction with rocd.
```

Comment:
I'd say it's more than just a UI for rocd. I see rocd as a much more general-purpose thing, allowing very different use-cases. roc-cast, OTOH, implements one or a few quite specific use cases (home audio & AoIP in buildings, mostly). So I expect that roc-cast will have business logic on top of rocd specific to its use-cases.